### PR TITLE
feat(sdk): added sanitizing k8s object process and test. Fixes: #332

### LINKED ISSUE #332: Sanitizing Kubernetes resources into strings
/kind feature

**Description:**
Some Kubernetes resources such as `V1EnvVar` for container environment variables only accept string, so we need to let the SDK to sanitize these variables into proper strings before dumping it to Tekton YAML.

The Kubernetes API docs defines what types are supported under the pod spec.
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/

/good-first-issue

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by jizg**
**Which issue is resolved by this Pull Request:** 
Resolves #332

**Description of your changes:**
Added process and testcases for recursively sanitizing k8s container object type based on the kubernetes.client.models definition. Currently primitive types of string, integer, boolean, list of string are all the types included in kubernetes.client.models.V1Container and its children attributes.

**Environment tested:**

* Python Version (use `python --version`): 3.7.7
* Tekton Version (use `tkn version`): 0.13.1
* Kubernetes Version (use `kubectl version`): v1.16.15
* OS (e.g. from `/etc/os-release`): Ubuntu 18.04

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by jizg** [comment]
@googlebot I signed it!

**Comment by jizg** [comment]
@Tomcli How to sign CLA? Shall choose `contribution on behalf of your Employer`?

**Comment by drewbutlerbb4** [comment]
Hi @jizg there is a process that you need to follow to join the IBM Google CLA group before you are covered under the CLA. I have sent you a slack message with the IBM internal link to the instructions.

**Comment by Tomcli** [comment]
Thanks @jizg for your contribution. This PR looks good to me. We can approve this PR once you signed the cla.

**Comment by jizg** [comment]
I have raised a request to join IBM contributor google group, and it should be soon to sign the CLA.

**Comment by jizg** [comment]
@googlebot I signed it!

**Comment by jizg** [comment]
@googlebot I signed it!

**Comment by jizg** [comment]
@Tomcli I've signed the CLA. Would you please check whether it's ok to merge this PR? Thanks

File: sdk/python/kfp_tekton/compiler/_k8s_helper.py [hunk 1]
```diff
@@ -115,3 +115,90 @@ def convert_k8s_obj_to_json(k8s_obj):
 
     return {key: convert_k8s_obj_to_json(val)
             for key, val in iteritems(obj_dict)}
+
+
+def _to_str(s):
+    return None if s is None else str(s)
```
Comment (drewbutlerbb4) [inline sdk/python/kfp_tekton/compiler/_k8s_helper.py:121]: Thanks @jizg, quick question. Is there a reason we need to check for `None` here? Doesn't this get checked in sanitize_k8s_object? Or are there plans to use `_to_str ()` outside this file in the future?

Comment (jizg) [reply sdk/python/kfp_tekton/compiler/_k8s_helper.py]: You are right, '_to_str()' is not necessary now. I firstly did the 'None' type check in '_to_str()', and then moved the check to `sanitize_k8s_object` but forgot to remove '_to_str()'. I have updated the codes. Thanks.

**Comment by drewbutlerbb4** [comment]
Thanks @jizg 
/lgtm

**Comment by Tomcli** [comment]
thanks @jizg 
/lgtm
/approve

**Pull request merged**